posthog/api/notebook.py

```diff
@@ -0,0 +1,178 @@
+        user=user,
+        item_id=notebook_id,
+        scope="Notebook",
+        activity=activity,
+        detail=Detail(changes=changes, short_id=notebook_short_id),
+    )
+
+
+class NotebookSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Notebook
+        fields = [
+            "id",
+            "short_id",
+            "title",
+            "content",
+            "deleted",
+            "created_at",
+            "created_by",
+            "last_modified_at",
+            "last_modified_by",
+        ]
+        read_only_fields = [
+            "id",
+            "short_id",
+            "created_at",
+            "created_by",
+            "last_modified_at",
+            "last_modified_by",
+        ]
+
+    created_by = UserBasicSerializer(read_only=True)
+    last_modified_by = UserBasicSerializer(read_only=True)
+
+    def create(self, validated_data: Dict, *args, **kwargs) -> Notebook:
+        request = self.context["request"]
+        team = self.context["get_team"]()
+
+        self._check_can_create_notebook(team)
+
+        created_by = validated_data.pop("created_by", request.user)
+        notebook = Notebook.objects.create(
+            team=team, created_by=created_by, last_modified_by=request.user, **validated_data
+        )
+
+        log_notebook_activity(
+            activity="created",
+            notebook_id=notebook.id,
+            notebook_short_id=str(notebook.short_id),
+            organization_id=self.context["request"].user.current_organization_id,
+            team_id=team.id,
+            user=self.context["request"].user,
+        )
+
+        return notebook
+
+    def update(self, instance: Notebook, validated_data: Dict, **kwargs) -> Notebook:
+        try:
+            before_update = Notebook.objects.get(pk=instance.id)
+        except Notebook.DoesNotExist:
+            before_update = None
+
+        if validated_data.keys():
+            instance.last_modified_at = now()
+            instance.last_modified_by = self.context["request"].user
+
+        updated_notebook = super().update(instance, validated_data)
+        changes = changes_between("Notebook", previous=before_update, current=updated_notebook)
+
+        log_notebook_activity(
+            activity="updated",
+            notebook_id=str(updated_notebook.id),
+            notebook_short_id=str(updated_notebook.short_id),
+            organization_id=self.context["request"].user.current_organization_id,
+            team_id=self.context["team_id"],
+            user=self.context["request"].user,
+            changes=changes,
+        )
+
+        return updated_notebook
+
+    @staticmethod
+    def _check_can_create_notebook(team: Team) -> bool:
+        notebook_count = Notebook.objects.filter(deleted=False, team=team).count()
+        if notebook_count > 10:
+            raise PermissionDenied("You have hit the limit for notebooks for this team. Delete some to create more.")
+        return True
```

Comment:
I'm really not sure that we would ever limit number of notebooks tbh. Maybe we remove this until we actually have a concrete idea what this would look like?